pkg/network/server.go

```diff
@@ -499,13 +569,40 @@ func (s *Server) handleMessage(peer Peer, msg *Message) error {
 				return err
 			}
 			go s.startProtocol(peer)
+
+			s.tryStartConsensus()
 		default:
 			return fmt.Errorf("received '%s' during handshake", msg.CommandType())
 		}
 	}
 	return nil
 }
 
+func (s *Server) handleNewPayload(p *consensus.Payload) {
+	h := p.Hash()
+	for peer := range s.Peers() {
+		if !peer.Handshaked() {
+			continue
+		}
+		payload := payload.NewInventory(payload.ConsensusType, []util.Uint256{h})
+		s.RelayDirectly(peer, payload)
+	}
+}
+
+func (s *Server) requestTx(hashes ...util.Uint256) {
+	if len(hashes) == 0 {
+		return
+	}
+
+	for peer := range s.Peers() {
+		if !peer.Handshaked() {
+			continue
+		}
+		payload := payload.NewInventory(payload.TXType, hashes)
+		s.RelayDirectly(peer, payload)
+	}
```

Comment:
This looks similar to `handleNewPayload`, I think we can refactor out some common part (and maybe it's the same for relaying code).